tests/functional/test_thermal_heuristic_one_cluster.py

```diff
@@ -0,0 +1,358 @@
+        - Demand is constant to 2000 MW except for the 13th hour for which it is 2050 MW
+        - Thermal generation is characterized with:
+            - P_min = 700 MW
+            - P_max = 1000 MW
+            - Min up time = 3
+            - Min down time = 10
+            - Generation cost = 50€ / MWh
+            - Startup cost = 50
+            - Fixed cost = 1 /h
+            - Number of unit = 3
+        - Unsupplied energy = 1000 €/MWh
+        - Spillage = 0 €/MWh
+
+    The optimal solution consists in having 3 units turned on between time steps 10 and 19 with production equal to 2100 to respect pmin and 2 the rest of the time. Fast heuristic turns on 3 units for 10 timesteps because min down time is equal to 10.
+
+    The optimal cost is then :
+          50 x 2000 x 158 (prod step 1-9 and 20-168)
+        + 50 x 2100 x 10 (prod step 10-19)
+        = 16 850 000
+    """
+
+    number_hours = 168
+
+    thermal_problem_builder = ThermalProblemBuilder(
+        fast=True,
+        data_dir=Path(__file__).parent / data_path,
+        id_thermal_cluster_model=THERMAL_CLUSTER_MODEL_MILP.id,
+        port_types=[BALANCE_PORT_TYPE],
+        models=[FastModelBuilder(THERMAL_CLUSTER_MODEL_MILP).model] + models,
+        time_scenario_hour_parameter=TimeScenarioHourParameter(1, 1, 168),
+    )
+
+    cluster = thermal_problem_builder.heuristic_components()[0]
+
+    # First optimization
+    resolution_step_1 = thermal_problem_builder.main_resolution_step(
+        week_scenario_index
+    )
+
+    thermal_problem_builder.update_database_heuristic(
+        resolution_step_1.output,
+        week_scenario_index,
+        list_cluster_id=None,
+        var_to_read="generation",
+        param_to_update="n_guide",
+        fn_to_apply=lambda x, y: ceil(round(x / y, 12)),
+        param_needed_to_compute=["p_max"],
+    )
+    # Solve heuristic problem
+    resolution_step_heuristic = thermal_problem_builder.heuristic_resolution_step(
+        id_component=cluster,
+        index=week_scenario_index,
+        model=HeuristicFastModelBuilder(
+            number_hours, delta=compute_delta(cluster, thermal_problem_builder.database)
+        ).model,
+    )
+    thermal_problem_builder.update_database_heuristic(
+        resolution_step_heuristic.output,
+        week_scenario_index,
+        None,
+        var_to_read="n",
+        param_to_update="min_generating",
+        fn_to_apply=lambda x, y, z: min(x * y, z),
+        param_needed_to_compute=["p_min", "max_generating"],
+    )
+
+    for time_step in range(number_hours):
+        assert (
+            thermal_problem_builder.database.get_value(
+                ComponentParameterIndex(cluster, "min_generating"), time_step, 0
+            )
+            == 3 * 700
+            if time_step in [t for t in range(10, 20)]
+            else 2 * 700
+        )
+
+    # Second optimization with lower bound modified
```

Comment:
Hard to see while reading the code where the bound modification has happened, I guess it is in `update_database_heuristic`, this is why the advice on wrapping this function